webapp/src/components/backstage/playbook_runs/playbook_run/retrospective.tsx

```diff
@@ -1,13 +1,258 @@
-import React from 'react';
+import React, {useEffect, useRef, useState} from 'react';
 import styled from 'styled-components';
+import {useIntl} from 'react-intl';
+import {DateTime} from 'luxon';
+import debounce from 'debounce';
 
-const Retrospective = () => {
-    return <Container>{'Retrospective'}</Container>;
+import {PlaybookRun, RunMetricData} from 'src/types/playbook_run';
+import {PlaybookWithChecklist} from 'src/types/playbook';
+import {clientFetchPlaybook, publishRetrospective, updateRetrospective} from 'src/client';
+import {useAllowPlaybookAndRunMetrics, useAllowRetrospectiveAccess} from 'src/hooks';
+import UpgradeBanner from 'src/components/upgrade_banner';
+import {AdminNotificationType} from 'src/constants';
+import {Timestamp} from 'src/webapp_globals';
+import {AnchorLinkTitle, Content, ELAPSED_TIME} from 'src/components/backstage/playbook_runs/shared';
+import MetricsData from '../playbook_run_backstage/metrics/metrics_data';
+import Report from '../playbook_run_backstage/retrospective/report';
+import ConfirmModalLight from 'src/components/widgets/confirmation_modal_light';
+import {PrimaryButton} from 'src/components/assets/buttons';
+
+interface Props {
+    playbookRun: PlaybookRun;
+    setPlaybookRun: (value: React.SetStateAction<PlaybookRun | null>) => void
+}
+
+const editDebounceDelayMilliseconds = 2000;
```

Comment:
optional: constant style

```suggestion
const DEBOUNCE_2_SECS = 2000;
```